cmd/skopeo/sync.go

```diff
@@ -0,0 +1,509 @@
+	if len(c.Args()) != 1 {
+		cli.ShowCommandHelp(c, "sync")
+		return errors.New("Exactly one argument expected")
+	}
+
+	if c.IsSet("source") && c.IsSet("source-file") {
+		return fmt.Errorf(
+			"Cannot use the '--source' and '--source-file' flags at the same time")
+	}
+
+	if !c.IsSet("source") && !c.IsSet("source-file") {
+		return fmt.Errorf(
+			"Must specify either a '--source' or a '--source-file'")
+	}
+
+	policyContext, err := getPolicyContext(c)
+	if err != nil {
+		return fmt.Errorf("Error loading trust policy: %v", err)
+	}
+	defer policyContext.Destroy()
+
+	destRef, err := getImageReference(c.Args()[0])
+	if err != nil {
+		return fmt.Errorf("Error while parsing destination: %v", err)
+	}
+	destinationCtx, err := contextFromGlobalOptions(c, "dest-")
+	if err != nil {
+		return err
+	}
+
+	signBy := c.String("sign-by")
+	removeSignatures := c.Bool("remove-signatures")
+
+	//TODO: should we assume that's our default manifest type?
+	manifestType := manifest.DockerV2Schema2MediaType
+
+	var srcRepoList []repoDescriptor
+
+	if c.IsSet("source") {
+		srcRepo, err := syncSourceHandler(c, destRef)
+		if err != nil {
+			return err
+		}
+		srcRepoList = append(srcRepoList, srcRepo)
+	} else {
+		srcRepoList, err = syncSourceFileHandler(c)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, srcRepo := range srcRepoList {
+		options := copy.Options{
+			RemoveSignatures:      removeSignatures,
+			SignBy:                signBy,
+			ReportWriter:          os.Stdout,
+			DestinationCtx:        destinationCtx,
+			ForceManifestMIMEType: manifestType,
+			SourceCtx:             srcRepo.Context,
+		}
+
+		for counter, ref := range srcRepo.TagsReferences {
+			logrus.WithFields(logrus.Fields{
+				"repo": ref.DockerReference().Name(),
+			}).Infof("Processing tag %d/%d", counter+1, len(srcRepo.TagsReferences))
+			destRef, err := buildFinalDestination(srcRepo.RootReference, ref, c.Args()[0])
+			if err != nil {
+				return err
+			}
+			logrus.WithFields(logrus.Fields{
+				"source":      ref,
+				"destination": destRef,
+			}).Info("Copy started")
+
+			err = copy.Image(context.Background(), policyContext, destRef, ref, &options)
+			if err != nil {
+				logrus.WithFields(logrus.Fields{
+					"repo": ref.DockerReference().Name(),
+				}).Error("Error copying tag, skipping")
+				logrus.Debug(err)
+				continue
+			}
+		}
+	}
+
+	return nil
+}
+
+var syncCmd = cli.Command{
+	Name:  "sync",
+	Usage: "Sync one or more images from one location to another",
+	Description: fmt.Sprint(`
+
+	Copy all the images from SOURCE to DESTINATION.
+
+	Useful to keep in sync a local docker registry mirror. Can be used
+	to populate also registries running inside of air-gapped environments.
+
+	SOURCE can be either a repository hosted on a docker registry
+	(eg: docker://docker.io/busybox) or a local directory
+	(eg: dir:///media/usb/).
+
+	SOURCE-FILE is a YAML file with a set of source images from different
+	docker registry. Local directory are not supported.
```

Comment:
Suggest:  
SOURCE-FILE is a YAML file with a set of source images from a different

Then either:
docker registry. Local directory is not supported.

or 
docker registry. Local directories are not supported.

I prefer the second.